src/easylink/step.py

```diff
@@ -1210,6 +1226,92 @@ def _validate(self) -> None:
         if errors:
             raise ValueError("\n".join(errors))
 
+    def validate_step(
+        self,
+        step_config: LayeredConfigTree,
+        combined_implementations: LayeredConfigTree,
+        input_data_config: LayeredConfigTree,
+    ) -> dict[str, list[str]]:
+        """Validates the ``Step`` assigned to this ``EmbarrassinglyParallelStep``.
+
+        Parameters
+        ----------
+        step_config
+            The internal configuration of this ``Step``, i.e. it should not include
+            the ``Step's`` name.
+        combined_implementations
+            The configuration for any implementations to be combined.
+        input_data_config
+            The input data configuration for the entire pipeline.
+
+        Returns
+        -------
+            A dictionary of errors, where the keys are the ``Step`` name and the
+            values are lists of associated error messages.
+
+        Notes
+        -----
+        If the ``EmbarrassinglyParallelStep`` does not validate (i.e. errors are
+        found and the returned dictionary is non-empty), the tool will exit and
+        the pipeline will not run.
+
+        We attempt to batch error messages as much as possible, but there may be
+        times where the configuration is so ill-formed that we are unable to handle
+        all issue in one pass. In these cases, new errors may be found after the
+        initial ones are handled.
+        """
+        return self.step.validate_step(
+            LayeredConfigTree(step_config),
+            combined_implementations,
+            input_data_config,
+        )
+
+    def set_configuration_state(
+        self,
+        step_config: LayeredConfigTree,
+        combined_implementations: LayeredConfigTree,
+        input_data_config: LayeredConfigTree,
+    ):
+        """Sets the configuration state to 'non-leaf'.
+
+        In addition to setting the configuration state, this also updates the
+        :class:`~easylink.graph_components.StepGraph` and
+        :class:`SlotMappings<easylink.graph_components.SlotMapping>`.
+
+        Parameters
+        ----------
+        step_config
+            The internal configuration of this ``Step``, i.e. it should not include
+            the ``Step's`` name.
+        combined_implementations
+            The configuration for any implementations to be combined.
+        input_data_config
+            The input data configuration for the entire pipeline.
+        """
+        if self.name != self.step.name:
+            # Update the slot mappings if the children have been renamed
```

Comment:
Why is the child step getting renamed? This logic looks good but I don't understand when this would be needed

Reply:
Added a comment. Basically a TemplatedStep parent gets renamed when unraveling